tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,486 @@
+from typing import Dict, List, Optional, ClassVar, Set
+
+import pytest
+from unittest import mock
+
+from great_expectations.core.domain import Domain
+from great_expectations.core.metric_domain_types import MetricDomainTypes
+from great_expectations.rule_based_profiler.config import ParameterBuilderConfig
+from great_expectations.rule_based_profiler.parameter_builder import (
+    ParameterBuilder,
+)
+from great_expectations.types.attributes import Attributes
+from great_expectations.rule_based_profiler.parameter_container import (
+    ParameterContainer,
+    FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY,
+    FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY,
+    FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY,
+    get_fully_qualified_parameter_names,
+)
+
+
+"""
+Tests in this module focus on behavior aspects of "ParameterBuilder.build_parameters()" -- this public method assesses
+dependencies on evaluation "ParameterBuilder" objects (if any specified), and resolves these dependencies, prior to
+calling its own interface method, "ParameterBuilder.build_parameters()".  One key method that is under test is
+"ParameterBuilder.resolve_evaluation_dependencies()" in terms of how it affects contents of shared memory object --
+"parameters: Dict[str, ParameterContainer]", whose "ParameterContainer" stores computation results of every
+"ParameterBuilder", evaluated within scope of given "Domain" object.  When one "ParameterBuilder" specifies dependencies
+on other "ParameterBuilder" objects for its evaluation, then "ParameterBuilder.resolve_evaluation_dependencies()" will
+process these dependencies recursively.  Hence, bebefore/after tests are provided for behavior of this method.  Utility
+method "get_fully_qualified_parameter_names()", whose thoroughly tests are elsewhere, is used to query shared memory.
```

Comment:
is there an opportunity now to move tests for other parts of parameter builder into unit tests here? Tests for a single module should not be elsewhere